docs/rollout.md

```diff
@@ -0,0 +1,175 @@
+# Rollout Strategy in Fleet
+
+Fleet uses a rollout strategy to control how apps are deployed across clusters. This feature allows you to define the order and grouping of cluster deployments, enabling controlled rollouts, and safer updates. Fleet evaluates the **Ready** status of each `BundleDeployment` to determine when to proceed to the next group. For more information, refer to [Status Fields](ref-status-fields.md).
+
+For initial deployments:
+
+* One cluster may be shown as **NotReady**.
+
+* Remaining clusters appear as **Pending**.
+
+For rollouts:
+
+* One cluster may be **NotReady**.
+
+* Remaining clusters are marked **OutOfSync** until the current one becomes ready
+
+Fleet creates up to 50 BundleDeployments at a time,  regardless of the rollout configuration. After creating a batch, Fleet evaluates the deployment status. If the current batch meets readiness criteria, Fleet proceeds to the next batch. Otherwise, it pauses the rollout.
+
+The rollout behavior is configured in the [`rolloutStrategy` in the `fleet.yaml`](ref-fleet-yaml.md)
+
+## How Does Partitioning Work?
+
+Partitions are considered non-ready if any cluster in the partition is not Ready. Deployment continues only if the number of non-ready partitions remains below maxUnavailablePartitions.
+
+:::note
+Fleet deploys up to 50 clusters at a time regardless of partition size.
+:::
+
+![A visual asset displaying flow of rollout in Fleet.](../static/img/flow-rollout-fleet.png)
+
+Various limits that can be configured in Fleet:
+
+| Field | Description | Default |
+| ----- | ----- | ----- |
+| maxUnavailable | Maximum number or percentage of clusters that can be non-ready before halting rollout. | 100% |
+| maxUnavailablePartitions | Number or percentage of partitions that can be non-ready at once. | 0 |
+| autoPartitionSize | Number or percentage of clusters per auto-created partition. | 25% |
+| partitions | Define manual partitions by cluster labels or group. If set, autoPartitionSize is ignored. | – |
+
+Fleet supports both automatic and manual rollout partitioning. For more information, refer to the [rolloutStrategy in  fleet.yaml.](ref-fleet-yaml.md)
+
+**Automatic Partitioning:** Fleet automatically creates partitions using autoPartitionSize.
+
+* If fewer than 200 clusters, Fleet uses a single partition.  
+* If 200+ clusters, partitions are created based on autoPartitionSize.
+
+**Manual Partitioning:** You define specific partitions using the partitions field. This gives you fine-grained control over cluster groupings and rollout order.
+
+:::note
+If you specify partitions manually, the autoPartitionSize is ignored.
+:::
+
+Partitions can be specified by name, `cluster selector`, `cluster group`, and `clusterGroupSelector`.
+
+For example, consider creating a partition named canary for production clusters in a specific group:
```

Comment:
I suggest we don't use the word "canary" at all. As far as I understand it, it means something that we do not support (yet). Using it might give people the wrong idea of us supporting it.